feat(pipes-targets): add step functions state machine target (#29987)
aws/aws-cdk: packages/@aws-cdk/aws-pipes-targets-alpha/README.md

Pipe targets are the end point of a EventBridge Pipe.

The following targets are supported:

1. `targets.SqsTarget`: [Send event source to a Queue](#amazon-sqs)

> 2. `targets.StateMachineTarget`: [Start a State Machine from an event source](#aws-step-functions)

Reviewer: Pending the code rename
```suggestion
2. `targets.SfnStateMachine`: [Invoke a State Machine from an event source](#aws-step-functions)
```

    target: pipeTarget
});
```

> ### AWS Step Function

Reviewer: ```suggestion
### AWS Step Functions
```

Author: Good catch! Changed it

    target: pipeTarget
});
```

> ### AWS Step Functions

Reviewer: This makes sense as a heading. It could also be
```suggestion
### AWS Step Functions State Machine
```

A State Machine can be used as a target for a pipe. The State Machine will be invoked with the (enriched/filtered) source payload.

```ts
declare const sourceQueue: sqs.Queue;
declare const targetStateMachine: sfn.IStateMachine;

const pipeTarget = new targets.StateMachineTarget(targetStateMachine,{});

const pipe = new pipes.Pipe(this, 'Pipe', {
    source: new SomeSource(sourceQueue),
    target: pipeTarget
});
```

> Specifying the Invocation Type of the State Machine.

Reviewer: This was already good. Slightly more explicit:
```suggestion
Specifying the Invocation Type when the target State Machine is invoked:
```

```ts
declare const sourceQueue: sqs.Queue;
declare const targetStateMachine: sfn.IStateMachine;

const pipeTarget = new targets.StateMachineTarget(targetStateMachine,
    {
      invocationType: targets.StepFunctionInvocationType.FIRE_AND_FORGET,
    }
);

const pipe = new pipes.Pipe(this, 'Pipe', {
    source: new SomeSource(sourceQueue),
    target: pipeTarget
});
```

> The target input can be transformed:

Reviewer: ```suggestion
The input to the target State Machine can be transformed:
```
